plugins/modules/zos_volume_init.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) IBM Corporation 2022
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+module: zos_volume_init
+short_description: Initialize volumes or minidisks on z/OS through ICKDSF init command.
+description:
+  - Wraps the ICKDSF INIT command to initialize volume or minidisk.
+  - Writes volume label and VTOC on volume or minidisk.
+version_added: 1.6.0
+author:
+  - "Austen Stewart (@stewartad)"
+  - "Almigdad Suliman (@Almigdad-Suliman)"
+  - "Nicholas Teves (@TODO)"
+  - "Nuoya Xie (@nxie13)"
+  - "Trevor Glassey (@tkglassey)"
+  - "Tyler Edwards (@TLEdwards-Git)"
+  - "Ketan Kelkar (@ketankelkar)"
+
+options:
+  volume_address:
+    description:
+      - 3 or 4 hexadecimal digit address of the volume to initialize.
+      - This could be a unit address or device number.
+    required: true
+    type: str
+  verify_existing_volid:
+    description:
+      - Verify that the provided volume serial matches the one found on existing volume/minidisk.
+      - Module fails if volser does not match.
+      - Note - This option is NOT a boolean, please leave it blank in order to skip the verification.
+    required: false
+    type: str
+  verify_offline:
+    description:
+      - Verify that the device is offline to all other systems.
+      - Beware, defaults set on target z/OS systems may override ICKDSF parameters.
+    type: bool
+    required: false
+    default: true
+  volid:
+    description:
+      - Specify the volume serial number to initialize the volume with.
+      - Expects 1-6 alphanumeric, national ($, \\#, \\@) or special characters.
+      - volid's specified with less than 6 characters are left justified and padded with blank chars (X'40').
+      - Characters are not validated so check with operating system guide for valid alphanumeric characters.
```

Comment:
I chose not to add validation because I did not think it wise to potentially restrict functionality -- figured it was ok to let ickdsf or zos handle problematic input in this case

Reply:
I read this as the only validation happening is the padding for a volid with less than 6 chars, is there more validation happening than this? If so, it might make sense to let ICKDSF propagate errors so long as they bubble up correctly. Do you have an example of this? If it is limited to just the length and padding, to me it makes seems we could fail that fast rather than the system log a system log error for something we can handle.

Reply:
Waiting on ICKDSF team to advise.

Reply:
the approach seems to be "if z/OS will allow it, ICKDSF will allow it", decision was to get rid of the "note" line:
~`Characters are not validated so check with operating system guide for valid alphanumeric characters.`~